app/controllers/application_controller.rb

```diff
@@ -5,6 +5,30 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+
+    enable :sessions unless test? #test? refers to base method used for environment selection
+    set :session_secret, "secret"
+  end
+
+  get '/' do
+    erb :'application/root'
+  end
+
+  helpers do
+
+    def logged_in?
+      #the '!!' double bang converts object into a truthy value statement
+      !!session[:user_id]
+    end
+
+    def current_user
```

Comment:
Because in the lifespan of a request, you might call `current_user` multiple times, we'd prefer this method to only hit the database once and then 'cache' the user instance into an instance variable that could be returned on any subsequent call to `current_user`.

```ruby
def current_user
  @current_user ||= User.find(session[:id]) if session[:id]
end
```

This pattern of conditionally setting an instance variable or returning it if it's already defined using the `||=` (or-equals) operator is called Memoization ([read more](http://gavinmiller.io/2013/basics-of-ruby-memoization/) and [here](http://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/)).